lib/internal/fs/rimraf.js

```diff
@@ -42,6 +42,34 @@ const separator = Buffer.from(sep);
 function rimraf(path, options, callback) {
   let retries = 0;
 
+  if (options.recursiveStrict) {
+    if (stat(path, (err, stats) => {
+      if (err) {
+        if (err.code === 'ENOENT') {
+          throw err;
+        } else if (!stats.isDirectory()) {
+          throw new ERR_INVALID_ARG_VALUE('path', path, 'is not a directory')
+        }
+      }
+
+      _rimraf(path, options, function CB(err) {
```

Comment:
Repeating this function seems bad. I could move it into an external function or something else?

Reply:
Yeah, I wonder if you could just pass `strict` down into the other helper, and vary the behavior accordingly?